packages/flame/lib/src/widgets/sprite_widget.dart

```diff
@@ -106,7 +106,10 @@ class InternalSpriteWidget extends StatelessWidget {
   @override
   Widget build(_) {
     return Container(
-      child: CustomPaint(painter: SpritePainter(sprite, anchor, angle: angle)),
+      child: CustomPaint(
+        painter: SpritePainter(sprite, anchor, angle: angle),
+        size: sprite.srcSize.toSize(),
```

Comment:
This will prevent the user to render the sprite bigger than its original size right?

I think that a better approach here would be to use the [BoxFit](https://docs-flutter-io.firebaseapp.com/flutter/painting/BoxFit-class.html) enum. Or, maybe something like that. wdyt?

Reply:
Yeah, I get that, but this will solve one issue, while adding a new one, where all sprite widgets will always have the size of the image. And it will be impossible to render it bigger than its original size.

So we need an alternative here.

Reply:
So, i'm looking at `RenderCustomPaint` render-box, and it says that it considers the provided size as the _preferred_ size, meaning it will attempt to find the size that is as close as possible to the given size, while still satisfying the parent's constraints. Meaning that if constrains say that minSize=maxSize=100, then that will be size of the widget regardless of what its `size` property says.

I've tried this with the SpriteWidgetExample, and it can still scale the sprite both up and down.

Reply:
> So, i'm looking at `RenderCustomPaint` render-box, and it says that it considers the provided size as the _preferred_ size, meaning it will attempt to find the size that is as close as possible to the given size, while still satisfying the parent's constraints. Meaning that if constrains say that minSize=maxSize=100, then that will be size of the widget regardless of what its `size` property says.
> 
> I've tried this with the SpriteWidgetExample, and it can still scale the sprite both up and down.

Interesting! So my concern will not happen! Thanks for checking on that. LGTM!